### DISCUSSION TITLE
https://imagekit.io/docs/integration/nextjs

### DISCUSSION THREAD

**Discussion by giscus** [imagekitio/discussions#56] [Docs]
# https://imagekit.io/docs/integration/nextjs

Real-time image & video resizing, automatic optimization, and file uploading in Next.js using ImageKit.io.

https://imagekit.io/docs/integration/nextjs

**Comment by acatzk** [comment]
Do you have a support for node version 18.12.0?

**Reply by acatzk** [reply]
I just copied all your source code and it's still working

**Reply by imagekitio** [reply]
Should work. We are working on the issue created of SDK repo on GitHub. Hopefully, we will be able to release a new version soon.

**Reply by acatzk** [reply]
Yeah, Good to hear!

**Reply by ahnv** [reply]
Hey @acatzk,

We've updated the SDK to work on all Node.js versions  >= 12 with imagekit-next 1.0.1.

This reply is a bit overdue, so please note that newer releases are available now. If you haven’t updated yet, you might want to grab the latest version for the best experience.

Let us know if you run into any issues!

**Comment by rahul2008d** [comment]
I have a question: Can I upload images in the background? I already have the image file and want to pass it to the " IKUpload " component without it prompting for the image file, so that it uploads the same file.

**Reply by imagekitio** [reply]
For such custom things, you should consider using our Javascript SDK directly. It provides an `upload` function. Docs - https://github.com/imagekit-developer/imagekit-javascript?tab=readme-ov-file#file-upload

**Reply by imagekitio** [reply]
Checkout 2.0.0 - https://imagekit.io/docs/integration/nextjs#uploading-files

**Comment by JBEnterprises** [comment]
Hey all. What's the general consensus on using sizes and srcSet the context of this package?

**Reply by imagekitio** [reply]
For now you can do it using loader as explained here https://imagekit.io/docs/integration/nextjs#responsive-images. In future, we will add support for automatically generating srcset based on sizes.

**Reply by imagekitio** [reply]
We just released 2.0.0 that handles responsive images out of the box https://imagekit.io/docs/integration/nextjs#responsive-images

**Comment by divya-nsh** [comment]
How to handle cases where an image is uploaded from the frontend by user, but he never clicks on the save button or change the image? The image will not remain unused know it should be deleted or marked to be deleted.

**Reply by imagekitio** [reply]
ImageKitClient exposes methods from ImageKit javascript SDK. You can use `upload` method to upload the file in custom code.

**Reply by imagekitio** [reply]
Checkout 2.0.0 - https://imagekit.io/docs/integration/nextjs#uploading-files

**Comment by KNIGHTCORE47** [comment]
hey imagekit, in nextjs do i have to download both @imagekit/next and imagekit as well to delete any media from imagekit over sdk, or there is a method in v2 like upload ...?

**Reply by imagekitio** [reply]
To delete files, you will have to use Node.js SDK in your Next project.
